File: src/main/java/com/noktiz/ui/web/behavior/DisableOnClassBehavior.js
/* 
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

function DisableOnClassBehavior(componentId, htmlClass, enable) {
    id=componentId+"_"+htmlClass+"_"+enable;
    if (typeof enableMap === "undefined") {
        enableMap = new Object();
    }
    if(typeof enableMap[enable]==="undefined"){
        enableMap[enable]=true;
    }
    if (typeof DisableOnClassCallId === 'undefined') {
        DisableOnClassCallId = new Object();
    }
    if(typeof DisableOnClassCallId[id]!== "undefined"){
        return;
    }
    callid=setInterval( function() {
        var classes = jQuery("#" + componentId).attr('class').split(" ");
        var found = 0;
        for (i = 0; i < classes.length; i++) {
            if (classes[i] === htmlClass) {
                found++;
            }
        }
        if (!found) {
            enableMap[enable] = true;
        }
        else {
            enableMap[enable] = false;
        }
    },100);
    DisableOnClassCallId[id] = callid;
}

File: src/main/java/com/noktiz/ui/web/behavior/DisableOnClassBehavior.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.noktiz.ui.web.behavior;

import org.apache.wicket.Component;
import org.apache.wicket.behavior.Behavior;
import org.apache.wicket.markup.head.IHeaderResponse;
import org.apache.wicket.markup.head.JavaScriptHeaderItem;
import org.apache.wicket.markup.head.OnDomReadyHeaderItem;
import org.apache.wicket.request.resource.PackageResourceReference;

/**
 *
 * @author Hossein
 */
public class DisableOnClassBehavior extends  Behavior{
    String enable;
    String htmlClass;
    public DisableOnClassBehavior(String enable,String htmlClass) {
        this.enable = enable;
        this.htmlClass=htmlClass;
    }
    
    @Override
    public void beforeRender(Component component) {
        component.setOutputMarkupId(true);
    }

    @Override
    public void renderHead(Component component, IHeaderResponse response) {
        response.render(JavaScriptHeaderItem.forReference(new PackageResourceReference(DisableOnClassBehavior.class, "DisableOnClassBehavior.js")));
        String markupId = component.getMarkupId();
        StringBuilder sb= new StringBuilder();
        sb.append("DisableOnClassBehavior('").append(markupId).append("','").
                append(htmlClass).append("','").
                append(enable).append("');");
        response.render(OnDomReadyHeaderItem.forScript(sb.toString()));
    }
    
}
